src/txkube/_network.py

```diff
@@ -219,7 +217,7 @@ class _NetworkClient(object):
     def _request(self, method, url, headers=None, bodyProducer=None):
         action = start_action(
             action_type=u"network-client:request",
-            method=method.decode("ascii"),
+            method=ensure_unicode(method),
```

Comment:
`method` looks like it's always bytes to me.  If it isn't maybe it's the caller that's at fault.

Reply:
I thought that if I made this helper function, you would want me to use it everywhere that `.decode()` was used.

I have been using test-driven development to do this work, but it is a bit difficult, because
there are still many broken tests on Python 3, and some of the fixes are intertwined with other fixes.  For example there are things I fixed in https://github.com/LeastAuthority/txkube/pull/162 , but haven't been fixed in this branch, so show up as failures.

Please clarify the direction that I should go in on this branch.  If https://github.com/LeastAuthority/txkube/pull/162 and this branch (in some form you find acceptable) are merged into master, that will make things easier to move forward with.

Thanks.

Reply:
It doesn't make sense to me to use this anywhere that isn't already following the pattern `if isinstance(s, bytes): s.decode()` ... since that's what `ensure_unicode` does.

Using `isinstance` to check if something is `bytes` before decoding it when we already _know_ the thing is bytes doesn't improve anything.  It's slower and it introduces uncertainty to the maintainer whether maybe it _isn't_ always bytes.

If something is bytes, just call decode on it if you want a str.

Reply:
I removed this in the updated patch.